expenses-manager-frontend/src/main/frontend/src/app/add-expense-dialog/add-expense-dialog.component.html

```diff
@@ -9,14 +9,14 @@ <h1><fa-icon [icon]="['fas', 'money-bill']"></fa-icon> Add expense</h1>
       <div class="date-icon">
         <span class="icons"><i class="glyphicon glyphicon-calendar"></i></span>
       </div>
-      <input type="date" class="form-control input-date" onclick="this.showPicker()" id="dateInput" aria-label="date-input">
+      <input type="date" class="form-control input-date" [(ngModel)]="selectedDate" [ngModelOptions]="{standalone: true}" onclick="this.showPicker()" id="dateInput" aria-label="date-input" required>
```

Comment:
Pentru toate aceste field-uri decat sa legi cu ngModel la o variabile recomand sa integram toate field-urile intr-un tag de form:
**<form [formGroup]="form" (ngSubmit)="onSubmit()">** iar fiecare input va fi legat la cate un field folosind **formControlName** ( <input type="date" class="form-control input-date" formControlName="selectedDate" aria-label="date-input" required>) . In fisierul de typescript vom folosi FormGroup pentru form-ul mare, si FormGroup pentru fiecare field in parte.

 form: FormGroup;
 constructor(private fb: FormBuilder) {}

  ngOnInit(): void {
    // Definim FormGroup și controalele sale
    this.form = this.fb.group({
      selectedDate: ['', Validators.required],
      selectedCategory: ['', Validators.required],
      selectedAmount: [0, Validators.required],
      description: ['']
    });
}
onSubmit() {
    if (this.form.valid) {
      console.log(this.form.value);
      // Aici poți procesa datele din formular
    }
  }